### PR TITLE
Fix tags Bug

### PR DESCRIPTION
**Pull request by lafama**
Allow tag case insensitive search
List tags using Title case

### CODE REVIEW

**Comment by chrisallenlane** [comment]
@lafama, thanks for reaching out, and thanks for using `wit-cms`.

I'm respectfully going to close this without merging (I'll explain why), but I think I have an alternative solution to your problem. Let's discuss a few things.

From your PR:
> List tags using Title case

I didn't want to do that simply because I don't believe that every user is going to want that. (I personally don't want that on my own site, for one.) `wit-cms` was designed to allow users to use whatever capitalization they prefer when writing tag and category names.

That was the idea, anyway.

Your PR made me to realize there was a bug in the prior implementation, though: while a post's `category` and `tags` properties previously supported mixed case, it turns out that the _routes_ for accessing posts by tag and category were actually lowercasing parameters in the URL before looking up posts, thus making posts containing mixed case irretrievable by tag or category. ([Here is the relevant diff][diff].)

That bug has been fixed in `wit-cms` version `4.1.1`, which has been published to npm.

Now, with regards to your problem specifically:

Now that the above is fixed, you should be able to simply change the `tags` and `categories` properties to use title case directly in your post front-matter, and everything should work as expected.

TL;DR: you didn't need a case-insensitive implementation - you just needed the case-sensitive implementation to work properly.

Please let me know if you have any questions or comments.

[diff]: https://github.com/chrisallenlane/wit-cms/commit/5ab2c648d3b2d104364b229ed2d37bb07e31ce2b

**Comment by lafama** [comment]
Thats perfect,
My worry is  two different posts with the same Tag but with different case 
Example : 'Staff' tag would be treated differently with 'staff' yet they should be generally the same tag

**Comment by chrisallenlane** [comment]
Gotcha.

So currently, all posts should be retrievable, but they will still be treated separately.

For example, let's say you have two tags: `"foo"` and `"Foo"`. You'll now be able to retrieve the appropriate posts by their tag (whereas previously `"Foo"` would have been irretrievable due to the bug), but you won't be able to get the `"Foo"` and `"foo"` posts in a single query.

One thing to consider:

If that still poses a problem for you, know that title-case can actually be applied in CSS using `text-transform: capitalize`. ([Reference][mdn]) With that said, it might make sense only to use lowercase characters in your tag and category front-matter, and then to handle the "presentation" in CSS.

I hope that helps!

[mdn]: https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform
